include/ofi_net.h

```diff
@@ -184,15 +286,24 @@ ofi_byteq_write(struct ofi_byteq *byteq, const void *buf, size_t len)
 void ofi_byteq_writev(struct ofi_byteq *byteq, const struct iovec *iov,
 		      size_t cnt);
 
-static inline ssize_t ofi_byteq_recv(struct ofi_byteq *byteq, SOCKET sock)
+static inline ssize_t ofi_byteq_recv(struct ofi_byteq *byteq, SOCKET sock,
+		      struct ofi_uring_ctx *uctx)
 {
 	size_t avail;
 	ssize_t ret;
 
 	avail = ofi_byteq_writeable(byteq);
 	assert(avail);
-	ret = ofi_recv_socket(sock, &byteq->data[byteq->tail], avail,
-			      MSG_NOSIGNAL);
+#ifdef HAVE_LIBURING
```

Comment:
almost every place `ofi_uring_initialized` is called it is preceded by `#ifdef HAVE_LIBURING`.  Rather than having these preprocessor checks throughout the code, can we have a `ofi_uring_initialized` implementation that always returns false, if HAVE_LIBURING is false/undefined?  I believe the compiler should be able to interpret that and optimize the branches

Reply:
ifdefs within a function should definitely be avoided.  It makes reading and maintaining the code more difficult.  ifdefs should basically almost always go surround an entire function.

Reply:
Yes, I agree with you both. I believe the code to wrap io_uring functions (e.g., `ofi_uring_recv()`) should be declared in `include/linux/osd.h` instead of ` include/ofi_net.h`. OSD code for other operating systems (Windows, BSD, OSX) would have the same io_uring functions declared but would return -FI_ENOSYS. What do you think?